File: C#/primitive_array.cs
using System;

class DynamicArray
{
    private int[] array;
    private int size;

    public DynamicArray(int capacity = 2)
    {
        array = new int[capacity];
        size = 0;
    }

    public void Add(int value)
    {
        if (size == array.Length)
        {
            int[] newArray = new int[array.Length * 2];
            Array.Copy(array, newArray, array.Length);
            array = newArray;
        }
        array[size++] = value;
    }

    public void Print()
    {
        for (int i = 0; i < size; i++)
        {
            Console.Write(array[i] + " ");
        }
        Console.WriteLine();
    }
}

class Program
{
    static void Main()
    {
        DynamicArray arr = new DynamicArray();
        arr.Add(1);
        arr.Add(2);
        arr.Add(3);
        arr.Add(4);
        arr.Print();
    }
}

File: Python/primitive_array.py
class DynamicArray:
    def __init__(self, capacity=2):
        self.array = [0] * capacity
        self.size = 0

    def add(self, value):
        if self.size == len(self.array):
            new_array = [0] * (len(self.array) * 2)
            for i in range(len(self.array)):
                new_array[i] = self.array[i]
            self.array = new_array
        self.array[self.size] = value
        self.size += 1

    def print(self):
        print(self.array[:self.size])

arr = DynamicArray()
arr.add(1)
arr.add(2)
arr.add(3)
arr.add(4)
arr.print()
